python/lsst/multiprofit/plotting/plot_model_rgb.py

```diff
@@ -0,0 +1,398 @@
+            (5, 95),
+        )
+        if rgb_min_auto:
+            kwargs["minimum"] = rgb_min
+        if rgb_stretch_auto:
+            kwargs["stretch"] = 2 * (rgb_max - rgb_min)
+
+    img_rgb = apVis.make_lupton_rgb(*images_data, **kwargs)
+    if has_model:
+        img_model_rgb = apVis.make_lupton_rgb(*images_model, **kwargs)
+    aspect = np.clip((y_max - y_min) / (x_max - x_min), 0.25, 4)
+
+    n_rows = 1 + has_model
+    n_cols_gs = 1 + has_model
+    n_cols_rgb = 1 + has_model * (1 + plot_chi_hist)
+    figsize_y = 8 * n_rows * aspect
+
+    fig_rgb, ax_rgb = plt.subplots(nrows=n_rows, ncols=n_cols_rgb, figsize=(8 * n_cols_rgb, figsize_y))
+    fig_gs, ax_gs = (
+        (None, None)
+        if not plot_singleband
+        else plt.subplots(
+            nrows=n_bands,
+            ncols=n_cols_gs,
+            figsize=(8 * n_cols_gs, 8 * aspect * n_bands),
+        )
+    )
+    (ax_rgb[0][0] if has_model else ax_rgb).imshow(img_rgb, extent=extent, origin="lower")
+    (ax_rgb[0][0] if has_model else ax_rgb).set_title("Data")
+    if has_model:
+        ax_rgb[1][0].imshow(img_model_rgb, extent=extent, origin="lower")
+        ax_rgb[1][0].set_title(f"Model ({band_str})")
+
+    masks_inv = {}
+    # Create a mask of high-sn pixels (based on the model)
+    mask_inv_highsn = np.ones(img_rgb.shape[:1], dtype="bool") if high_sn_threshold else None
+
+    for idx, band in enumerate(bands):
+        obs = observations[band]
+        mask_inv = obs.mask_inv.data
+        masks_inv[band] = mask_inv
+        img_data = obs.image.data
+        img_sigma_inv = obs.sigma_inv.data
+        if plot_singleband:
+            if has_model:
+                img_model = models[band]
+                if mask_inv_highsn:
+                    mask_inv_highsn *= (img_model * np.nanmedian(img_sigma_inv)) > high_sn_threshold
+                residual = (img_data - img_model) * mask_inv
+                value_max = np.nanpercentile(np.abs(residual), 98)
+                ax_gs[idx][0].imshow(residual, cmap="gray", vmin=-value_max, vmax=value_max, origin="lower")
+                ax_gs[idx][0].tick_params(labelleft=False)
+                ax_gs[idx][0].set_title(f"{band}-band Residual (abs.)")
+                ax_gs[idx][1].imshow(
+                    np.clip(residual * img_sigma_inv, -chi_max, chi_max),
+                    cmap="gray",
+                    origin="lower",
+                )
+                ax_gs[idx][1].tick_params(labelleft=False)
+                ax_gs[idx][1].set_title(f"{band}-band Residual (chi, +/- {chi_max:.2f})")
+            else:
+                ax_gs[idx].imshow(img_data * mask_inv * (img_sigma_inv > 0), cmap="gray", origin="lower")
+                ax_gs[idx].set_title(band)
+
+    if has_model:
+        # TODO: Draw masks in each channel? or total?
+        # mask_inv_all = np.prod(list(masks_inv.values()), axis=0)
```

Comment:
Remove comment?

Reply:
I'm leaving it there as a reminder for when I come across a better idea for overlaying masks on RGB images. I guess for the grayscale images I can shade them some colour as we normally do.